autoupdate/update.py

```diff
@@ -0,0 +1,420 @@
+            if cert.certificate_type == CertificateType.ROOT
+        )
+        logging.info(
+            "Root certificate will be %s: %s",
+            "created" if root_certificate.filename in added_certificates else "updated",
+            root_certificate.filename,
+        )
+
+        test_root_certificate = next(
+            cert
+            for cert in self.certificates
+            if cert.certificate_type == CertificateType.TEST_ROOT
+        )
+        logging.info(
+            "Test root certificate will be %s: %s",
+            "created"
+            if test_root_certificate.filename in added_certificates
+            else "updated",
+            test_root_certificate.filename,
+        )
+
+        intermediate_certificates = sorted(
+            [
+                cert
+                for cert in self.certificates
+                if cert.certificate_type == CertificateType.INTERMEDIATE
+            ],
+            key=lambda c: c.filename,
+        )
+        logging.info(
+            "%d intermediate certificates to be updated:",
+            len(intermediate_certificates),
+        )
+        for certificate in intermediate_certificates:
+            logging.info(
+                " %s %s",
+                "+" if certificate.filename in added_certificates else "~",
+                certificate,
+            )
+
+        test_certificates = sorted(
+            [
+                cert
+                for cert in self.certificates
+                if cert.certificate_type == CertificateType.TEST
+            ],
+            key=lambda c: c.filename,
+        )
+        logging.info("%d test certificates to be updated:", len(test_certificates))
+        for certificate in test_certificates:
+            logging.info(
+                " %s %s",
+                "+" if certificate.filename in added_certificates else "~",
+                certificate,
+            )
+
+        if not self.dry_run:
+            self._perform_update_directory(directory_path, removed_certificates)
+
+    def _perform_update_directory(
+        self, directory_path: str, removed_certificates: set[str]
+    ):
+        logging.info("Performing update")
+
+        for certificate in self.certificates:
+            logging.info("Updating certificate %s", certificate.filename)
+            with (pathlib.Path(directory_path) / certificate.filename).open("wb") as f:
+                f.write(requests.get(certificate.link).content)
+
+        for certificate in removed_certificates:
+            logging.info("Removing certificate %s", certificate)
+            (pathlib.Path(directory_path) / certificate).unlink()
+
+    def update_constants(self, constants_path: str):
+        logging.info("Updating constants file %s", constants_path)
+
+        test_certs = repr(
+            {
+                cert.title: cert.filename
+                for cert in self.certificates
+                if cert.certificate_type != CertificateType.TEST
```

Comment:
I'm guessing this should be:
```suggestion
                if cert.certificate_type == CertificateType.TEST
```

Reply:
Yup, that is what is causing the oddities with LIVE_CERTS containing test certificates and vice-versa.